src/commands/deploy.ts

```diff
@@ -51,47 +53,74 @@ $ bump deploy FILE --dry-run --doc <doc_slug> --token <your_doc_token>
-    const action = flags['dry-run'] ? 'validate' : 'deploy';
-    /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-    const [documentation, token] = [flags.doc!, flags.token!];
 
-    this.d(`${args.FILE} looks like an ${api.specName} spec version ${api.version}`);
+    const [dryRun, documentation, token, hub, autoCreate, documentationName, branch] = [
+      flags['dry-run'],
+      /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+      flags.doc!,
+      /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+      flags.token!,
+      flags.hub,
+      flags['auto-create'],
+      flags['doc-name'],
+      flags.branch,
+    ];
+    const file = await statSync(args.FILE);
+
+    if (file.isDirectory()) {
+      throw new Error(
+        'Deploy with a directory is not yet implemented. Please wait while we are working on it!',
+      );
+    } else {
+      await this.deploySingleFile(
+        args.FILE,
+        dryRun,
+        documentation,
+        token,
+        hub,
+        autoCreate,
+        documentationName,
+        branch,
+      );
+    }
+
+    return;
+  }
+
+  private async deploySingleFile(
+    file: string,
+    dryRun: boolean,
+    documentation: string,
+    token: string,
+    hub: string | undefined,
+    autoCreate: boolean,
+    documentationName: string | undefined,
+    branch: string | undefined,
+  ): Promise<void> {
+    const action = dryRun ? 'validate' : 'deploy';
     cli.action.start(`* Let's ${action} a new documentation version on Bump`);
 
-    const request: VersionRequest = {
+    const response: VersionResponse | undefined = await new CoreDeploy(this.config).run(
+      file,
+      dryRun,
       documentation,
-      hub: flags.hub,
-      documentation_name: flags['doc-name'],
-      auto_create_documentation: flags['auto-create'] && !flags['dry-run'],
-      definition,
-      references,
-      branch_name: flags.branch,
-    };
-
-    const response = flags['dry-run']
-      ? await this.bump.postValidation(request, token)
-      : await this.bump.postVersion(request, token);
+      token,
+      hub,
+      autoCreate,
+      documentationName,
+      branch,
+    );
 
     cli.action.stop();
 
-    switch (response.status) {
-      case 200:
-        cli.styledSuccess('Definition is valid');
-        break;
-      case 201:
-        const version: VersionResponse = response.data
-          ? response.data
-          : { id: '', doc_public_url: 'https://bump.sh' };
-
-        cli.styledSuccess(
-          `Your new documentation version will soon be ready at ${version.doc_public_url}`,
-        );
-        break;
-      case 204:
-        this.warn('Your documentation has not changed');
-        break;
+    if (response) {
+      cli.styledSuccess(
+        `Your new documentation version will soon be ready at ${response.doc_public_url}`,
+      );
+    } else if (dryRun) {
```

Comment:
Does this change mean something has changed in API?

If I understand correctly, before the change, API response code was used to distinguish dry-mode or not, and now it's only based on `dryRun` variable (ie flag `'dry-run'`).

Reply:
Before this change, we mixed the type of HTTP responses into a single variable. Now we only read the return value of the `CoreDeploy.run(...)` “service”.

If `dryRun` was `true` we would send a validation request ([`POST /api/v1/validations`](https://developers.bump.sh/operation/operation-post-validations#operation-post-validations-responses)) which only possible **valid** response was a `200` http code. So if `CoreDeploy.run(...)` returns an `undefined` value it means “everything went well”.

If `dryRun` was `false` we would send a version creation request ([`POST /api/v1/versions`](https://developers.bump.sh/operation/operation-post-versions#operation-post-versions-responses)) which have two possible responses a `201` or a `204`. So if `CoreDeploy.run(...)` returns a `VersionResponse` it means the deploy is ongoing, if it returns a `undefined` it means the documentation has not changed and nothing needs to be done.

I can change the order of if it's clearer for you something like:

```js
if (dryRun) {
  // Definition is valid
} else {
  if (response) {
    // Deploy is ongoing
  } else {
    // Doc not changed
  }
}
```

Would that help?

Reply:
Suggestion is maybe a bit clearer (especially when we read comment above), but what really helps is the sync discussion we just had :) Thanks!